boards/PDM/Src/App/App_Efuse.c

```diff
@@ -1,17 +1,207 @@
+}
+
+float App_Efuse_GetChannel1Current(struct Efuse *efuse)
+{
+    return efuse->get_channel_1_current(efuse->io_efuse);
+}
+
+bool App_Efuse_Channel0_CurrentLowCheck(struct Efuse *efuse)
+{
+    return efuse->get_channel_0_current(efuse->io_efuse) < efuse->channel_0_min_current;
+}
+
+bool App_Efuse_Channel0_CurrentHighCheck(struct Efuse *efuse)
+{
+    return efuse->get_channel_0_current(efuse->io_efuse) > efuse->channel_0_max_current;
+}
+
+bool App_Efuse_Channel1_CurrentLowCheck(struct Efuse *efuse)
+{
+    return efuse->get_channel_1_current(efuse->io_efuse) < efuse->channel_1_min_current;
+}
+
+bool App_Efuse_Channel1_CurrentHighCheck(struct Efuse *efuse)
+{
+    return efuse->get_channel_1_current(efuse->io_efuse) > efuse->channel_1_max_current;
+}
+
+int App_Efuse_FaultProcedure_Channel0(struct Efuse *efuse, int max_attempts)
+{
+    static TimerChannel *efuse_timer;
+    static int           num_attempts      = 0;
+    static bool          has_timer_started = false;
+    App_Timer_InitTimer(efuse_timer, 1000);
+
+    if ((App_Efuse_GetChannel0Current(efuse) <= EFUSE_CURRENT_THRESHOLD && App_Efuse_IsChannel0Enabled(efuse)) ||
+        App_Efuse_Channel0_CurrentHighCheck(efuse))
+    {
+        if (!has_timer_started)
+        {
+            App_Timer_Restart(efuse_timer);
+            has_timer_started = true;
+        }
+    }
+    else
+    {
+        App_Timer_Stop(efuse_timer);
+        has_timer_started = false;
+        num_attempts      = 0;
+        return 0; // fine
+    }
+
+    if (has_timer_started && App_Timer_UpdateAndGetState((efuse_timer)) == TIMER_STATE_EXPIRED)
+    {
+        App_Efuse_StandbyReset(efuse);
+        num_attempts++;
+    }
+    if (num_attempts >= max_attempts)
+    {
+        App_Timer_Stop(efuse_timer);
+        has_timer_started = false;
+        return 2; // channel failed
+    }
+
+    return 1; // unknown status
+}
+
+int App_Efuse_FaultProcedure_Channel1(struct Efuse *efuse, int max_attempts)
+{
+    static TimerChannel *efuse_timer;
+    static int           num_attempts      = 0;
+    static bool          has_timer_started = false;
+    App_Timer_InitTimer(efuse_timer, 1000);
+
+    if ((App_Efuse_GetChannel1Current(efuse) <= EFUSE_CURRENT_THRESHOLD && App_Efuse_IsChannel1Enabled(efuse)) ||
+        App_Efuse_Channel1_CurrentHighCheck(efuse))
+    {
+        if (!has_timer_started)
+        {
+            App_Timer_Restart(efuse_timer);
+            has_timer_started = true;
+        }
+    }
+    else
+    {
+        App_Timer_Stop(efuse_timer);
+        has_timer_started = false;
+        num_attempts      = 0;
+        return 0;
+    }
+
+    if (has_timer_started && App_Timer_UpdateAndGetState((efuse_timer)) == TIMER_STATE_EXPIRED)
+    {
+        App_Efuse_StandbyReset(efuse);
+        num_attempts++;
+    }
+    if (num_attempts >= max_attempts)
+    {
+        App_Timer_Stop(efuse_timer);
+        has_timer_started = false;
+        return 2;
+    }
+
+    return 1; // unknown status
```

Comment:
You should also make an enum of the different return values here since 0/1/2 isn't that clear. Also if the only checks are fault or no fault you could return a bool.